core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -76,6 +76,17 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
     Map("clientId" -> clientId)
   )
 
+  newGauge("DeadFetcherThreadCount", {
```

Comment:
I guess the choice here is between local naming inconsistency or global naming inconsistency. I think I slightly prefer making the two metrics in this PR consistent, but could be persuaded otherwise if you feel strongly.

Reply:
I think I'd feel strongly about it in this case. Most of the metric names follow the camel case notation and I think it'd be good to keep that. In fact almost only the log cleaner differs so in fact I'd change that (and later on hopefully refactor the others too).

Reply:
Can we change this to `DeadThreadCount`? As with the other one, I think the context makes it clear that it refers to the fetcher threads.